### DISCUSSION TITLE
more results available, increase resultsize parameter

### DISCUSSION THREAD

**Discussion by vanessav5** [silverhack/monkey365#109] [General]
Hi,

I try to execute this command with the last monkey365 version:

$assets = Invoke-Monkey365 -ExportTo HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Analysis ExchangeOnline

But I obtained the following message and the report didn't generate:

ADVERTENCIA: [16:26:37:327] - [Get-PSExoAdminApiObject] - There are more results available than are currently displayed. To view them, increase the
value for the ResultSize parameter. - warning - WST-XXXXX-

I'd like to obtain the report, I appreciate your help.

Regards,

Vanessa

**Comment by silverhack** [comment]
Hi @vanessav5, 

That's a very rare issue. Could you please add the -verbose -debug and -InformationAction Continue to your command? 

`Invoke-Monkey365 -ExportTo HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Analysis ExchangeOnline -Debug -Verbose -InformationAction Continue`

Seems a permission issue, but with Debug and Verbose options you'll see direct console output and will get you with detailed errors in case the report cannot be generated. 

Gracias,

**Reply by vanessav5** [reply]
Hi @silverhack 

Generating this report takes a lot of time, but I could catch this information from the logs:  
-------------------------------
DETALLADO: [20:11:06:424] - [Invoke-MonkeyJob] - Invoked all Jobs, Collecting the last jobs that are running - verbose - WST-2129.xxxxx-
CONSOLE: [20:11:06:428] - [Get-MonkeyRestEXOUser] - Task [4a3283e2-8148-45c0-963b-b0e12f927cde] - Getting Exchange Online users from xxx - info - WST-2129.xxx - ExoUsersInfo
DETALLADO: [20:11:06:435] - [New-HttpRequestMessage] - Adding X-SerializationLevel header - verbose - WST-2129.xxx -
DETALLADO: [20:11:06:438] - [New-HttpRequestMessage] - Adding Prefer header - verbose - WST-2129.xxx -
DETALLADO: [20:11:06:440] - [New-HttpRequestMessage] - Adding X-ResponseFormat header - verbose - WST-2129.xxx -
DETALLADO: [20:11:06:443] - [New-HttpRequestMessage] - Adding client-request-id header - verbose - WST-2129.xxx -
ADVERTENCIA: [20:11:20:026] - [Get-PSExoAdminApiObject] - There are more results available than are currently displayed. To view them, increase the value for the ResultSize parameter.
 - warning - WST-2129.xxx -
DETALLADO: [20:11:20:032] - [New-HttpRequestMessage] - Adding X-SerializationLevel header - verbose - WST-2129.xxx -
DETALLADO: [20:11:20:035] - [New-HttpRequestMessage] - Adding Prefer header - verbose - WST-2129.xxx -
DETALLADO: [20:11:20:039] - [New-HttpRequestMessage] - Adding X-ResponseFormat header - verbose - WST-2129.xxx -
DETALLADO: [20:11:20:041] - [New-HttpRequestMessage] - Adding client-request-id header - verbose - WST-2129.xxx -
ADVERTENCIA: [20:11:31:902] - [Get-PSExoAdminApiObject] - There are more results available than are currently displayed. To view them, increase the value for the ResultSize parameter.
 - warning - WST-2129.xxx -
DETALLADO: [20:11:32:045] - [New-HttpRequestMessage] - Adding X-SerializationLevel header - verbose - WST-2129.xxx -
DETALLADO: [20:11:32:049] - [New-HttpRequestMessage] - Adding Prefer header - verbose - [....EDITED....] -
DETALLADO: [20:11:32:052] - [New-HttpRequestMessage] - Adding X-ResponseFormat header - verbose - WST-2129.xxx -
DETALLADO: [20:11:32:055] - [New-HttpRequestMessage] - Adding client-request-id header - verbose - WST-2129.xxx -
DETALLADO: [20:11:40:792] - [Invoke-MonkeyJob] - Finishing the 1 job(s) that are still running - verbose - WST-2129.xxx -
DETALLADO: [20:11:41:569] - [Invoke-MonkeyJob] - Exiting script - verbose - WST-2129.xxx -
DETALLADO: [20:11:41:573] - [Invoke-MonkeyJob] - Jobs Collected: 1 - verbose - WST-2129.xxx -
DETALLADO: [20:11:41:576] - [Invoke-MonkeyJob] - Time took to Invoke and Complete the Jobs : 00:00:35.1528142 - verbose - WST-2129.xxx -
DETALLADO: [20:11:42:042] - [Invoke-MonkeyJob] - Invoked all Jobs, Collecting the last jobs that are running - verbose - WST-2129.xxx -
CONSOLE: [20:11:42:044] - [Get-MonkeyRestEXOUsersMailbox] - Task [0f563310-45df-4070-b5fe-82a879a876d5] - Getting Exchange Online Mailboxes from XXX - info - WST-2129.xxx - ExoMailboxesInfo
DETALLADO: [20:11:42:050] - [New-HttpRequestMessage] - Adding X-SerializationLevel header - verbose - WST-2129.xxx -
DETALLADO: [20:11:42:053] - [New-HttpRequestMessage] - Adding Prefer header - verbose - WST-2129.xxx -
DETALLADO: [20:11:42:055] - [New-HttpRequestMessage] - Adding X-ResponseFormat header - verbose - WST-2129.xxx -
DETALLADO: [20:11:42:058] - [New-HttpRequestMessage] - Adding client-request-id header - verbose - WST-2129.xxx -
ADVERTENCIA: [20:11:59:443] - [Get-PSExoAdminApiObject] - There are more results available than are currently displayed. To view them, increase the value for the ResultSize parameter.
 - warning - WST-2129.xxx -
DETALLADO: [20:11:59:450] - [New-HttpRequestMessage] - Adding X-SerializationLevel header - verbose - WST-2129.xxx -
DETALLADO: [20:11:59:452] - [New-HttpRequestMessage] - Adding Prefer header - verbose - WST-2129.xxx -
DETALLADO: [20:11:59:456] - [New-HttpRequestMessage] - Adding X-ResponseFormat header - verbose - WST-2129.xxx -
DETALLADO: [20:11:59:458] - [New-HttpRequestMessage] - Adding client-request-id header - verbose - WST-2129.xxx -
------------------------

I don't know if the obtained results are correct and what is the reason for taking a lot of time.

Regards,

VAnessa

**Reply by silverhack** [reply]
Hi @vanessav5, 

Your message was edited for privacy reasons. 

Regarding verbose output, and If no errors were found, then scan was finished correctly. 

On the other hand, the long time to finish is mainly because one reason. A very large environment with a potential Entra ID/Exchange Online with thousand of objects (users, groups, mailboxes, etc..).

Please, note that interactive credentials used with Monkey365 will time out and cause errors/exceptions. This is because Azure/Microsoft 365 access tokens [have an expiration](https://learn.microsoft.com/en-us/entra/identity-platform/access-tokens#token-lifetime) of roughly 60/90 minutes and Monkey365 does not currently support automatic refreshing tokens, so a workaround would be to execute Monkey365 with [service principal authentication](https://silverhack.github.io/monkey365/authentication/authFlows/sp/).
 So If your long run time is caused by a **large Microsoft 365 environment**, there is no current workaround. If you have large tenant, you can split the collection up by splitting services using the Analysis, IncludeEntraID and [ExcludeCollector](https://silverhack.github.io/monkey365/configuration/general-options/) flags.

Hope that helps, 
Saludos

**Reply by vanessav5** [reply]
Hi,

Oh, I understand.  Then I can ignore these warnings and consider the report is complete.

Thanks

Vanessa

**Reply by silverhack** [reply]
Hi, 

Yes. This warning message comes from your tenant and basically is telling you that the result query has more than 999 elements, and hence there are more results available. 

Monkey365 will iterate through the [odata.nextLink](https://docs.oasis-open.org/odata/odata-json-format/v4.0/cs01/odata-json-format-v4.0-cs01.html#_Annotation_odata.count) until a null value is returned. 

Hope that helps,
